README.md

```diff
@@ -41,8 +49,47 @@ Follow the steps to install the node packages required for both building and run
     ```
 2. To start the server, use the command `npm run start`
 3. The http server by default starts with 3000 port or whatever is declared in `process.env.PORT` (look at `index.ts` for more details)
-4. By default `get` call on `localhost:3000` gives a sample response
-5. Other routes include a `ping` with get and post. Make `get` or `post` request to `http://localhost:3000/ping`
+4. Health check available at path `health/ping` with get and post. Make `get` or `post` request to `http://localhost:3000/health/ping`
+
+
+## System flow
+---
+
+Diagram describes the Data service system flow
+
+```mermaid
+graph LR;
+    A[osw-validation] -->|subscribes| B(Data Service) -->|publishes| C(osw-data)
+    B -->|Save| D(OSW Database)
+    B -->|Auth| E(Auth Service)
+    G(Gateway) -->|GET| B(Data Service)
+```
+
+- `Data service`, subscribes to `osw-validation` topic to listen to validation results of the osw file upload request.
+
+- `Data service`, authorizes the request via `Auth Service` 
+
+- If validation result is failed , Data service publishes the information to `osw-data` topic to update request status complete without persisting the information.
```

Comment:
What "validation" are we talking about here?
We went directly from Authorization to "Validation result". Is it the auth validation? or what?